### PR TITLE
NEW: Modal and Portal component

### PR DESCRIPTION
**Pull request by wEprous**
This Pull Request meets the following criteria:

* [x] Tests have been added/adjusted for my new feature
* [x] New Components are registered in index.js of my project

### CODE REVIEW

File: src/Modal/ModalFooter/index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    padding: ${({ theme }) => `0 ${theme.orbit.spaceXXLarge} ${theme.orbit.spaceXXLarge}`};
+    border-bottom-left-radius: ${({ theme }) => theme.orbit.borderRadiusNormal};
+    border-bottom-right-radius: ${({ theme }) => theme.orbit.borderRadiusNormal};
+  `};
+
+  ${StyledChild}:last-of-type {
+    padding-right: 0;
+  }
+
+  ${media.mobileOnly`
```
Comment (tomashapl) [inline src/Modal/ModalFooter/index.js]: Please use mobileFirst and override

File: src/Modal/ModalFooter/index.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  background-color: ${({ theme }) => theme.orbit.paletteWhite};
+  // TODO: create token paddingModalFooter
+  padding: ${({ theme, fixed }) =>
+    fixed ? theme.orbit.spaceMedium : `0 ${theme.orbit.spaceMedium} ${theme.orbit.spaceMedium}`};
+  box-sizing: border-box;
+  box-shadow: ${({ fixed }) => fixed && `0 -2px 4px 0 rgba(23, 27, 30, 0.1)`};
+  // TODO: create token boxShadowActionableInverted
+
+  ${media.desktop`
+    position: static;
```
Comment (tomashapl) [inline src/Modal/ModalFooter/index.js]: `relative`

File: src/Modal/index.js.flow [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+type Children =
+  | React$Element<typeof ModalHeader>
+  | React$Element<typeof ModalSection>
+  | React$Element<typeof ModalFooter>;
+
+export type Props = {|
+  +size?: "small" | "normal" | "large",
+  +children: Array<Children> | Children,
+  +nonClosable?: boolean,
```
Comment (tomashapl) [inline src/Modal/index.js.flow]: `closable` with default `true`

File: src/Modal/index.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+    const contentIndent = React.Children.map(
+      children,
+      item => (item.type === ModalFooter ? item.props.fixed : false),
+    ).includes(true);
+
+    const mobileHeader = React.Children.map(
+      children,
+      item => (item.type === ModalHeader ? item.props.title : undefined),
+    ).toString();
```
Comment (tomashapl) [inline src/Modal/index.js]: can be passed as component, remove toString

File: src/Modal/index.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+
+    const close = (
```
Comment (tomashapl) [inline src/Modal/index.js]: refactor to Component

File: src/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+    if (ev.key === "Escape" && onClose) {
+      onClose(ev);
+    }
+  };
+
+  render() {
+    const { onClose, children, size = SIZES.MEDIUM, nonClosable = false } = this.props;
+    const { scrolled, loaded } = this.state;
+
+    const contentIndent = React.Children.map(
```
Comment (tomashapl) [inline src/Modal/index.js]: getIndend fci

File: src/Modal/index.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+      children,
+      item => (item.type === ModalFooter ? item.props.fixed : false),
+    ).includes(true);
+
+    const mobileHeader = React.Children.map(
```
Comment (tomashapl) [inline src/Modal/index.js]: getTitle = () =>

File: src/Modal/index.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+        <ButtonLink
+          onClick={onClose}
+          size="normal"
+          icon={<Close color="secondary" />}
+          transparent
+        />
+      </CloseContainer>
+    );
+
+    const inner =
```
Comment (tomashapl) [inline src/Modal/index.js]: refactor to Component

File: src/Modal/index.js [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+  box-sizing: border-box;
+  border-top-left-radius: ${({ theme }) => theme.orbit.borderRadiusNormal};
+  border-top-right-radius: ${({ theme }) => theme.orbit.borderRadiusNormal};
+  background-color: ${({ theme }) => theme.orbit.backgroundModal};
+  width: 100%;
+  min-height: calc(100% - ${({ theme }) => theme.orbit.spaceXLarge});
+  box-shadow: ${({ theme }) => theme.orbit.boxShadowModal};
+  overflow: hidden;
+
+  ${media.mobileOnly`
```
Comment (tomashapl) [inline src/Modal/index.js]: mobileFirst

File: src/Modal/ModalFooter/index.js (hunk 1 above)
Comment (wEprous) [reply src/Modal/ModalFooter/index.js]: This is the very edge case where I need to set `font-size` and `heigh` for mobile only. If I use the mobile first approach, I need to somehow redeclare these properties in `desktop`, but buttons can have `medium` and `large` sizes and you don't know what size was used.

File: src/Modal/index.js [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+const getTitle = children =>
+  React.Children.map(children, item => (item.type === ModalHeader ? item.props.title : undefined));
+
+const Content = ({ closable, onClose, size, children, scrolled }: ContentType) =>
+  closable && onClose ? (
+    <ClickOutside onClickOutside={onClose}>
+      <ModalWrapperContent size={size} contentIndent={getIndent(children)} scrolled={scrolled}>
+        <MobileHeader scrolled={scrolled}>
+          {getTitle(children)}
+          {closable && CloseElement(onClose)}
```
Comment (tomashapl) [inline src/Modal/index.js]: `<CloseElement onClose={onClose} />`

File: src/Modal/index.js [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+  `};
+`;
+
+MobileHeader.defaultProps = {
+  theme: defaultTokens,
+};
+
+const CloseElement = onClose => (
+  <CloseContainer>
+    <ButtonLink onClick={onClose} size="normal" icon={<Close color="secondary" />} transparent />
```
Comment (tomashapl) [inline src/Modal/index.js]: Please style hover state

**Pull request merged**